app/src/main/java/nl/tudelft/cs4160/trustchain_android/network/peer/Peer.java

```diff
@@ -1,150 +1,219 @@
 package nl.tudelft.cs4160.trustchain_android.network.peer;
 
-import java.io.ByteArrayInputStream;
-import java.io.ByteArrayOutputStream;
-import java.io.IOException;
-import java.io.ObjectInputStream;
-import java.io.ObjectOutputStream;
+import com.google.protobuf.ByteString;
+
 import java.io.Serializable;
 import java.net.InetAddress;
 import java.net.InetSocketAddress;
 import java.net.UnknownHostException;
-import java.nio.ByteBuffer;
 
+import nl.tudelft.cs4160.trustchain_android.crypto.PublicKeyPair;
 import nl.tudelft.cs4160.trustchain_android.main.OverviewConnectionsActivity;
+import nl.tudelft.cs4160.trustchain_android.message.MessageProto;
 
 /**
  * The peer object that is used to find other connected peers in the network.
- * The peer is identified by its unique peer id, which is the chosen username, and keeps track of the last send and receive time.
+ * The peer is identified by its unique public key pair, and keeps track of the last sent and received time.
  */
 public class Peer implements Serializable {
     final private static int TIMEOUT = 15000;
     final private static int REMOVE_TIMEOUT = 25000;
-    private InetSocketAddress address;
-    private String peerId;
-    private boolean hasReceivedData = false;
-    private boolean hasSentData = false;
-    private int connectionType;
-    private String networkOperator;
-    private long lastSendTime;
-    private long lastReceiveTime;
+    private MessageProto.Peer protoPeer;
+    private long lastSentTime = -1;
+    private long lastReceiveTime = -1;
     private long creationTime;
+    long serialVersionUID = -8872742022617111380L;
+
+    /**
+     * Create a peer, for consistency protocolbuffers is used for this too, instead of simply serializing this.
+     * @param address the address where the peer is located
+     * @param publicKeyPair the public key pair of the peer
+     * @param name  the nickname of the peer
+     */
+    public Peer(InetSocketAddress address, PublicKeyPair publicKeyPair, String name) {
```

Comment:
Yes, one is the peer object as it is used in messages, which holds the most important information like address and public key and the other is for local usage (because it's not possible to extend a protobuffer class) containing some additional methods and creation time etc. The protocolbuffer peer object you basically never interact with. This also allows for easier updates to the normal Peer class without serialization errors.